sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRelation.scala

```diff
@@ -88,7 +101,8 @@ private[sql] case class JDBCRelation(
 
   override val needConversion: Boolean = false
 
-  override val schema: StructType = JDBCRDD.resolveTable(url, table, properties)
+  override val schema: StructType =
+    JDBCRDD.resolveTable(url, table, JDBCRelation.getEffectiveProperties(properties))
```

Comment:
The non-serializability doesn't come into play here since this runs entirely on the driver, right? Or can this method wind up being called on executors?